### ISSUE TITLE
Upgrade to .Net 6 throws "System.ArgumentNullException : Value cannot be null. (Parameter 'provider')"

### ISSUE THREAD

**Issue by StevenRasmussen** [dotnet/efcore#26122]
## Bug or possible breaking change?

I tried to debug using the EF Core symbols and .Net Core symbols and it *seemed* like the error was originating from within `System.Linq.Queryable.Where` method [here](https://github.com/dotnet/runtime/blob/a7c69745a4e22757c69f65d2a750e2d6923e1186/src/libraries/System.Linq.Queryable/src/System/Linq/Queryable.cs#L42-L47)... but don't take my word for it ;)  That file originates in the .Net repo but my understanding is that EF Core should be providing the `provider` here.

### To reproduce:

1. Use the repo here: [https://github.com/StevenRasmussen/EFCore.SqlServer.NodaTime](https://github.com/StevenRasmussen/EFCore.SqlServer.NodaTime/tree/dot-net-6)
2. Make sure to use the branch called '**dot-net-6**'
3. Run any of the unit tests

### Stack trace

```
ServiceProviderServiceExtensions.GetRequiredService[T](IServiceProvider provider)
DbContextServices.CreateModel(Boolean designTime)
DbContextServices.get_Model()
<>c.<TryAddCoreServices>b__7_3(IServiceProvider p)
CallSiteRuntimeResolver.VisitFactory(FactoryCallSite factoryCallSite, RuntimeResolverContext context)
CallSiteVisitor`2.VisitCallSiteMain(ServiceCallSite callSite, TArgument argument)
CallSiteRuntimeResolver.VisitRootCache(ServiceCallSite callSite, RuntimeResolverContext context)
CallSiteRuntimeResolver.VisitScopeCache(ServiceCallSite callSite, RuntimeResolverContext context)
CallSiteVisitor`2.VisitCallSite(ServiceCallSite callSite, TArgument argument)
CallSiteRuntimeResolver.VisitConstructor(ConstructorCallSite constructorCallSite, RuntimeResolverContext context)
<79 more frames...>
DbContext.get_DbContextDependencies()
IDbContextDependencies.get_QueryProvider()
InternalDbSet`1.CreateEntityQueryable()
<>c.<get_EntityQueryable>b__11_0(InternalDbSet`1 internalSet)
NonCapturingLazyInitializer.EnsureInitialized[TParam,TValue](TValue& target, TParam param, Func`2 valueFactory)
InternalDbSet`1.get_EntityQueryable()
IQueryable.get_Provider()
Queryable.Where[TSource](IQueryable`1 source, Expression`1 predicate)
DurationQueryTests.Duration_can_be_used_in_query() line 21
```

### Provider and version information

EF Core version: 6.0.0-rc.1.21452.10
Database provider: Microsoft.EntityFrameworkCore.SqlServer
Target framework: .NET 6.0
Operating system: Windows 10
IDE: Visual Studio 2019 16.11.3 or Visual Studio 2022 17.0.0 Preview 4.1

**Comment by AndriySvyryd** [comment]
This is an intentional breaking change, many query services in EF are now registered as Scoped instead of Singleton, documentation is tracked by https://github.com/dotnet/EntityFramework.Docs/issues/3440

Generally you should only use `TryAddProviderSpecificServices` for the services that aren't added by EF or providers. Instead use `EntityFrameworkRelationalServicesBuilder.TryAdd` to register service replacements using the expected lifetime.

**Comment by StevenRasmussen** [comment]
@AndriySvyryd - Thanks!  Works like a charm... although I had to leave one of the calls to `TryAddProviderSpecificServices` for the `IMemberTranslatorProvider` as the default SQL Server provider was being used unless I changed that one back.  If you have any additional thoughts on that I'm open to suggestions.  See [here](https://github.com/StevenRasmussen/EFCore.SqlServer.NodaTime/blob/82d02bdb0f265cb218a3449cde8a5ddfc8b47aa8/src/SimplerSoftware.EntityFrameworkCore.SqlServer.NodaTime/Extensions/ServiceCollectionExtensions.cs#L21).
